lib/services/local/saml_idp_service_provider.go

```diff
@@ -164,6 +165,9 @@ func (s *SAMLIdPServiceProviderService) ensureEntityIDIsUnique(ctx context.Conte
 func validateSAMLIdPServiceProvider(sp types.SAMLIdPServiceProvider) error {
 	ed, err := samlsp.ParseMetadata([]byte(sp.GetEntityDescriptor()))
 	if err != nil {
+		if strings.Contains(err.Error(), "invalid url") {
```

Comment:
When is a ParseMetadata error not equivalent to BadParameter? Should we drop the if and wrap them all?